sic_framework/core/sic_logging.py

```diff
@@ -31,51 +32,50 @@ class SICRemoteError(Exception):
 
 
 class SICLogSubscriber(object):
-
+    """
+    A class to subscribe to a redis log channel, ensuring thread-safety with a mutex.
+    """
     def __init__(self):
         self.redis = None
-        # TODO make this a mutex
         self.running = False
+        self.logfile = open("sic.log", "w")
+        self.lock = threading.Lock()
 
-    def subscribe_to_log_channel_once(self):
+    def subscribe_to_log_channel(self):
         """
         Subscribe to the log channel and display any messages on the terminal to propagate any log messages in the
-        framework tot the user. This function may be called multiple times but will only subscribe once.
+        framework to the user. This function may be called multiple times but will only subscribe once.
         :return:
         """
-
-        if not self.running:
-            self.running = True
-            self.redis = SICRedis(parent_name="SICLogSubscriber")
-            self.redis.register_message_handler(
-                get_log_channel(), self._handle_log_message
-            )
+        with self.lock:  # Ensure thread-safe access
+            if not self.running:
+                self.running = True
+                self.redis = SICRedis(parent_name="SICLogSubscriber")
+                self.redis.register_message_handler(
+                    get_log_channel(), self._handle_log_message
+                )
 
     def _handle_log_message(self, message):
         """
-        Handle a message sent on a debug stream. Currently its just printed to the terminal.
+        Handle a message sent on a debug stream. Currently it's just printed to the terminal.
         :param message: SICLogMessage
         """
         print(message.msg, end="")
+        self.logfile.write(message.msg)
```

Comment:
The ANSI color codes are also written into the `log.file`.

Reply:
Fixed in https://github.com/Social-AI-VU/social-interaction-cloud/pull/30/commits/46b60f1062be39593e407c2f2b4c0b4b969c6867